ddprof-ffi/src/exporter.rs

```diff
@@ -275,8 +258,135 @@ pub unsafe extern "C" fn profile_exporter_send(
         Ok(HttpStatus(response.status().as_u16()))
     }() {
         Ok(code) => SendResult::HttpResponse(code),
-        Err(err) => SendResult::Failure(error_into_buffer(err)),
+        Err(err) => SendResult::Failure(err.into()),
+    }
+}
+
+#[export_name = "ddprof_ffi_SendResult_drop"]
+pub unsafe extern "C" fn send_result_drop(result: SendResult) {
+    std::mem::drop(result)
+}
+
+#[export_name = "ddprof_ffi_Vec_tag_new"]
+pub extern "C" fn vec_tag_new<'a>() -> crate::Vec<Tag<'a>> {
+    crate::Vec::default()
+}
+
+/// Pushes the tag into the vec.
+#[export_name = "ddprof_ffi_Vec_tag_push"]
+pub unsafe extern "C" fn vec_tag_push<'a>(vec: &mut crate::Vec<Tag<'a>>, tag: Tag<'a>) {
+    vec.push(tag)
+}
+
+#[allow(clippy::ptr_arg)]
+#[export_name = "ddprof_ffi_Vec_tag_as_slice"]
+pub extern "C" fn vec_tag_as_slice<'a>(vec: &'a crate::Vec<Tag<'a>>) -> Slice<'a, Tag<'a>> {
+    vec.as_slice()
+}
+
+#[export_name = "ddprof_ffi_Vec_tag_drop"]
+pub extern "C" fn vec_tag_drop(vec: crate::Vec<Tag>) {
+    std::mem::drop(vec)
+}
+
+struct ParseTagsError {
+    message: &'static str,
+}
+
+impl Debug for ParseTagsError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Tag Error: {}", self.message)
+    }
+}
+
+impl Display for ParseTagsError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "The tags string was malformed.")
+    }
+}
+
+impl Error for ParseTagsError {}
```

Comment:
My rust-foo is somewhat failing me here. Does this arrangement mean that by default we don't present a more specific error message, and only debug builds get it? If so, why? 😉

Reply:
The Rust convention is that `Debug` is for developers and `Display` is meant for everyone else. They are both present, and it depends on whether the programmer stringifies it with `{}` or `{:?}`.